buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -190,35 +190,34 @@ private void allocate(PoolThreadCache cache, PooledByteBuf<T> buf, final int req
 
             final PoolSubpage<T> head = table[tableIdx];
 
-            /**
-             * Synchronize on the head. This is needed as {@link PoolChunk#allocateSubpage(int)} and
-             * {@link PoolChunk#free(long)} may modify the doubly linked list as well.
-             */
-            synchronized (head) {
-                final PoolSubpage<T> s = head.next;
-                if (s != head) {
-                    assert s.doNotDestroy && s.elemSize == normCapacity;
-                    long handle = s.allocate();
-                    assert handle >= 0;
-                    s.chunk.initBufWithSubpage(buf, handle, reqCapacity);
-
-                    if (tiny) {
-                        allocationsTiny.increment();
-                    } else {
-                        allocationsSmall.increment();
+            try {
+                /**
+                 * Synchronize on the head. This is needed as {@link PoolChunk#allocateSubpage(int)} and
+                 * {@link PoolChunk#free(long)} may modify the doubly linked list as well.
+                 */
+                synchronized (head) {
+                    final PoolSubpage<T> s = head.next;
+                    if (s != head) {
+                        assert s.doNotDestroy && s.elemSize == normCapacity;
+                        long handle = s.allocate();
+                        assert handle >= 0;
+                        s.chunk.initBufWithSubpage(buf, handle, reqCapacity);
+                        return;
                     }
-                    return;
                 }
+                allocateNormal(buf, reqCapacity, normCapacity);
+                return;
+            } finally {
+                incTinySmallNormalAllocation(normCapacity);
```

Comment:
@isdom I would prefer if we not need to acquire the `synchronized(this)` two times when we increment for normal allocations. Can you please refactor to ensure we not do ?